lib/pid_file.rb

```diff
@@ -27,7 +27,7 @@ def remove
   def create(remove_on_exit = true)
     FileUtils.mkdir_p(File.dirname(@fname))
     File.open(@fname, "w") { |f| f.write(Process.pid) }
-    at_exit { PidFile.remove(@fname) } if remove_on_exit
+    at_exit { PidFile.remove(@fname) if PidFile.new(@fname).pid == Process.pid } if remove_on_exit
```

Comment:
Gah...I didn't realize this *was* PidFile class 😖 

Even so, what do you think of a different method to separate the concerns?